__tests__/main.runTaskListOnce.test.ts

```diff
@@ -225,6 +225,365 @@ test("supports future-scheduled jobs", () =>
     }
   }));
 
+test("allows update of pending jobs", () =>
+  withPgClient(async pgClient => {
+    await reset(pgClient);
+
+    const job1: Task = jest.fn(o => {
+      expect(o).toMatchObject({ a: "right" });
+    });
+    const tasks: TaskList = {
+      job1,
+    };
+
+    // Schedule a future job - note incorrect payload
+    const runAt = new Date();
+    runAt.setSeconds(runAt.getSeconds() + 3);
+
+    await pgClient.query(
+      `select graphile_worker.add_job('job1', '{"a": "wrong"}', run_at := '${runAt.toISOString()}', job_key := 'abc')`
+    );
+
+    // Assert that it has an entry in jobs / job_queues
+    const { rows: jobs } = await pgClient.query(
+      `select * from graphile_worker.jobs`
+    );
+    expect(jobs).toHaveLength(1);
+    const job = jobs[0];
+    expect(job.run_at).toEqual(runAt);
+
+    // Run all jobs (none are ready)
+    await runTaskListOnce(tasks, pgClient);
+    expect(job1).not.toHaveBeenCalled();
+
+    // update the job to run immediately with correct payload
+    const now = new Date();
+    await pgClient.query(
+      `select graphile_worker.add_job('job1', '{"a": "right"}', run_at := '${now.toISOString()}', job_key := 'abc')`
+    );
+
+    // Assert that it has updated the existing entry and not created a new one
+    const { rows: updatedJobs } = await pgClient.query(
+      `select * from graphile_worker.jobs`
+    );
+    expect(updatedJobs).toHaveLength(1);
+    const updatedJob = updatedJobs[0];
+    expect(updatedJob.run_at).toEqual(now);
+
+    // Run the task
+    await runTaskListOnce(tasks, pgClient);
+    expect(tasks.job1).toHaveBeenCalledTimes(1);
+  }));
+
+test("schedules a new job if existing is completed", () =>
+  withPgClient(async pgClient => {
+    await reset(pgClient);
+
+    const tasks: TaskList = {
+      job1: jest.fn(async () => {}),
+    };
+
+    // Schedule a job to run immediately
+    await pgClient.query(
+      `select graphile_worker.add_job('job1', '{"a": "first"}', job_key := 'abc')`
+    );
+
+    // run the job
+    await runTaskListOnce(tasks, pgClient);
+    expect(tasks.job1).toHaveBeenCalledTimes(1);
+
+    // attempt to update the job - it should schedule a new one instead
+    await pgClient.query(
+      `select graphile_worker.add_job('job1', '{"a": "second"}',  job_key := 'abc')`
+    );
+
+    // run again
+    await runTaskListOnce(tasks, pgClient);
+    expect(tasks.job1).toHaveBeenCalledTimes(2);
+
+    // check jobs ran in the right order
+    expect(tasks.job1).toHaveBeenNthCalledWith(
+      1,
+      { a: "first" },
+      expect.any(Object)
+    );
+    expect(tasks.job1).toHaveBeenNthCalledWith(
+      2,
+      { a: "second" },
+      expect.any(Object)
+    );
+  }));
+
+test("schedules a new job if existing is being processed", () =>
+  withPgClient(async pgClient => {
+    await reset(pgClient);
+
+    const tasks: TaskList = {
+      job1: jest.fn(async () => {
+        await sleep(50);
```

Comment:
Ok, I think I have this figured out. See https://github.com/graphile/worker/pull/63/commits/a970f16927771633a6f3f0ca71407ecbe464e244

I did have to add an extra sleep in the test body, because of the way the runner picks up a new job after the current one completes. 

Also, I can't see how to avoid a ts-ignore there - let me know if you have any ideas.